uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-balance-report.component.js

```diff
@@ -1,17 +1,114 @@
 /* eslint-env amd */
+/* global angular */
 
 define([
-  'leave-absences/shared/modules/components'
-], function (components) {
+  'common/lodash',
+  'leave-absences/shared/modules/components',
+  'leave-absences/shared/models/leave-balance-report.model',
+  'common/services/notification'
+], function (_, components) {
   components.component('leaveBalanceReport', {
-    controller: balanceReportController,
+    controller: controller,
     controllerAs: 'vm',
     templateUrl: ['shared-settings', function (sharedSettings) {
       return sharedSettings.sharedPathTpl + 'components/leave-balance-report.html';
     }]
   });
 
-  function balanceReportController () {
+  controller.$inject = ['$q', 'AbsenceType', 'LeaveBalanceReport', 'notification', 'Session'];
 
+  function controller ($q, AbsenceType, LeaveBalanceReport, notification, Session) {
+    var loggedInContactId;
+    var pageSize = 50;
+    var vm = this;
+
+    vm.absenceTypes = [];
+    vm.loading = { report: false };
+    vm.report = [];
+    vm.reportCount = 0;
+
+    /**
+     * Initializes the component. Loads logged in contact information and
+     * the first page of the report.
+     */
+    vm.$onInit = function () {
+      vm.loading.report = true;
+
+      $q.all([
+        initAbsenceTypes(),
+        initLoggedInContactId()
+      ])
+      .then(function () {
+        return vm.loadReportPage(1);
+      })
+      .catch(function (error) {
+        notification.error('Error', error);
+      });
+    };
+
+    /**
+     * Loads a specific page of the report.
+     *
+     * @return {Promise}
+     */
+    vm.loadReportPage = function (pageNumber) {
```

Comment:
I forgot to mention, but I made loadReportPage public so it can be used with the pagination component (next feature) like so:

`<pagination (params) on-page-change="vm.loadReportPage($page)"></pagination>`.

There are other alternatives like using $broadcast, but this is more straight forward. What do you think?